### PR TITLE
[EventHubs] GA doc preparation

### PR DESCRIPTION
**Pull request by yunhaoling**
azure-eventhub
- [ ] README
- [ ] CHANGELOG

checkpointstore-sync
- [ ] README
- [ ] CHANGELOG

checkpointstore-async
- [ ] README
- [ ] CHANGELOG

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/HISTORY.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Release History
 
-## 1.0.0b7 (Unreleased)
+## 1.0.0 (2020-01-06)
+This release is for version bump only. No feature changes.
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/HISTORY.md]: I would probably rephrase this to something like:
```
Stable release. No new features or API changes.
```

File: sdk/eventhub/azure-eventhub-checkpointstoreblob/HISTORY.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Release History
 
-## 1.0.0b7 (Unreleased)
+## 1.0.0 (2020-01-06)
+This release is for version bump only. No feature changes.
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub-checkpointstoreblob/HISTORY.md]: I would probably rephrase this to something like:
```
Stable release. No new features or API changes.
```

File: sdk/eventhub/azure-eventhub/HISTORY.md [hunk 3]
```diff
@@ -1,10 +1,26 @@
...
+## 5.0.0 (2020-01-06)
 
 **Breaking changes**
 
-- Removed deprecated property EventData.application_properties and deprecated method EventData.encode_message()
+- `EventData`
+    - Removed deprecated property `application_properties` and deprecated method `encode_message()`
+- `EventHubConsumerClient`
+    - `on_error` would be called when `EventHubConsumerClient` failed to claim ownership of partitions.
+    - `on_partition_cloe` and `on_partition_initialize` would be called in the case of meeting exceptions raised by callback `on_event`.
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/HISTORY.md]: Typo: `on_partition_close`
I would also change "meeting exceptions" to just "in the case of exceptions raised by `on_event` callback."

File: sdk/eventhub/azure-eventhub/HISTORY.md [hunk 4]
```diff
@@ -1,10 +1,26 @@
...
+        -  `EventHubConsumerClient` would close and re-open the internal partition receiver in this case.
+    - Default starting position from where `EventHubConsumerClient` should resume receiving when meeting an error has been re-prioritized.
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/HISTORY.md]: Minor: "resume receiving when meeting an error" -> "resume receiving after recovering from an error"

File: sdk/eventhub/azure-eventhub/HISTORY.md [hunk 5]
```diff
@@ -1,10 +1,26 @@
...
+        - If there is checkpoint, it will resume from the checkpoint.
+        - If there is no checkpoint but `starting_position` is provided, it will resume from `starting_posititon`.
+        - If there is no checkpoint and `starting_position`, it will resume from the latest position.
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/HISTORY.md]: "no checkpoint **_or_** `starting_postition`"

File: sdk/eventhub/azure-eventhub/HISTORY.md [hunk 6]
```diff
@@ -1,10 +1,26 @@
...
+- `get_partition_ids`, `get_partition_properties`, `get_eventhub_properties` would raise `AuthenticationException` when they fail to accomplish the request.
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/HISTORY.md]: See my comment on the earlier PR about whether AuthenticationException is the right error to raise here, in addition I would probably rephrase this to be more specific:
 "... will raise AuthenticationException when the service returns an error status code"

File: sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/HISTORY.md (hunk 1 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/HISTORY.md]: thanks, nice words!

File: sdk/eventhub/azure-eventhub-checkpointstoreblob/HISTORY.md (hunk 2 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub-checkpointstoreblob/HISTORY.md]: updated :P

File: sdk/eventhub/azure-eventhub/HISTORY.md (hunk 3 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/HISTORY.md]: updated

File: sdk/eventhub/azure-eventhub/HISTORY.md (hunk 4 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/HISTORY.md]: updated

File: sdk/eventhub/azure-eventhub/HISTORY.md (hunk 5 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/HISTORY.md]: updated

File: sdk/eventhub/azure-eventhub/HISTORY.md (hunk 6 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/HISTORY.md]: yes, let's update this after we settle down on that PR.

Comment (YijunXieMS) [reply sdk/eventhub/azure-eventhub/HISTORY.md]: In any case that place doesn't raise AuthenticationException, which is a uamqp exception. It's caught and are handled by _handle_error(), same as other uamqp exception.

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py [hunk 7]
```diff
@@ -58,8 +58,14 @@ class EventHubConsumerClient(ClientBase):
     :keyword float auth_timeout: The time in seconds to wait for a token to be authorized by the service.
      The default value is 60 seconds. If set to 0, no timeout will be enforced from the client.
     :keyword str user_agent: The user agent that should be appended to the built-in user agent string.
-    :keyword int retry_total: The total number of attempts to redo a failed operation when an error occurs. Default
-     value is 3.
+    :keyword int retry_total: The total number of attempts to redo a failed operation when an error occurs.
+     Default value is 3. The context of `retry_total` in receiving is special: The `receive` method is implemented
+     by a while-loop calling internal receive method in each iteration. In the `receive` case,
+     `retry_total` specifies the numbers of retry after error raised by internal receive method in the while-loop.
+     If retry attempts exhaust, the `on_error` callback would be called if provided carrying the error information,
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: "attempts are exhausted"
"would be called (ifprovided) with the error information. The failed internal partition consumer will be closed..."
"and a new internal partition consumer will be created...."

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py [hunk 8]
```diff
@@ -70,9 +76,11 @@ class EventHubConsumerClient(ClientBase):
      keys: `'proxy_hostname'` (str value) and `'proxy_port'` (int value).
      Additionally the following keys may also be present: `'username', 'password'`.
     :keyword checkpoint_store: Manager for storing the partition load-balancing and checkpoint data when receiving
-     events. If None, this `EventHubConsumerClient` instance will receive events without load-balancing and
-     checkpointing. The checkpoint store will be used in both cases of receiving from all partitions or a single
+     events. If not provided, `update_checkpoint` method `PartitionContext` object could still be called for in-memory
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: Not quite sure what this is saying.... perhaps something like this?
"For in-memory checkpointing operation, updating the checkpoint could (will? should?) happen automatically regardless of whether explicitly called in the supplied handler.".

Also - replace comma with period at end of line 80.

This comment also applies to the same doc change on line 188.

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py (hunk 7 above)
Comment (annatisch) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: This comment also applies to doc change on line 174

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py [hunk 9]
```diff
@@ -229,23 +244,28 @@ def receive(self, on_event, **kwargs):
          bool as the value indicating whether the starting_position for a specific partition is inclusive or not.
          This can also be a single bool value for all starting_position. The default value is False.
         :paramtype starting_position_inclusive: bool or dict[str,bool]
-        :keyword on_error: The callback function which would be called when there an error occurs during receiving.
+        :keyword on_error: The callback function which would be called when error is raised during receiving
+         after retry attempts exhaust and during the process of load balance.
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: The callback function that will be called when an error is raised during receiving 
after retry attempts are exhausted, or during the process of load-balancing.

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py [hunk 10]
```diff
@@ -229,23 +244,28 @@ def receive(self, on_event, **kwargs):
...
          The callback takes two parameters: `partition_context` which contains partition information
-         and `error` being the exception. The callback should be defined like so: `on_error(partition_context, error)`.
+         and `error` being the exception. `partition_context` could be None if the error is raised during
+         the process of load balance. The callback should be defined like: `on_error(partition_context, error)`.
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: "load-balancing"

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py [hunk 11]
```diff
@@ -229,23 +244,28 @@ def receive(self, on_event, **kwargs):
...
         :keyword on_partition_initialize: The callback function that will be called after a consumer for a certain
-         partition finishes initialization. The callback takes a single parameter: `partition_context`
+         partition finishes initialization. It would also be called when a new internal partition consumer is created
+         to take over the receiving process for a failed and closed internal partition consumer.
+         The callback takes a single parameter: `partition_context`
          which contains the partition information. The callback should be defined
-         like so: `on_partition_initialize(partition_context)`.
-        :paramtype on_partition_initialize: Callable[[~azure.eventhub.PartitionContext]]
+         like: `on_partition_initialize(partition_context)`.
+        :paramtype on_partition_initialize: Callable[[~azure.eventhub.aio.PartitionContext]]
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: This is not the async consumer, so should point to sync docs.

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py [hunk 12]
```diff
@@ -229,23 +244,28 @@ def receive(self, on_event, **kwargs):
...
         :keyword on_partition_close: The callback function that will be called after a consumer for a certain
-         partition is closed. The callback takes two parameters: `partition_context` which contains partition
-         information and `reason` for the close. The callback should be defined like so:
+         partition is closed. It would be also called when error is raised during receiving after retry attempts
+         exhaust. The callback takes two parameters: `partition_context` which contains partition
+         information and `reason` for the close. The callback should be defined like:
          `on_partition_close(partition_context, reason)`.
          Please refer to :class:`CloseReason<azure.eventhub.CloseReason>` for the various closing reasons.
-        :paramtype on_partition_close: Callable[[~azure.eventhub.PartitionContext, ~azure.eventhub.CloseReason]]
+        :paramtype on_partition_close: Callable[[~azure.eventhub.aio.PartitionContext, ~azure.eventhub.CloseReason]]
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: Same comment re aio docs

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py [hunk 13]
```diff
@@ -229,23 +244,28 @@ def receive(self, on_event, **kwargs):
...
          which contains the partition information. The callback should be defined
-         like so: `on_partition_initialize(partition_context)`.
-        :paramtype on_partition_initialize: Callable[[~azure.eventhub.PartitionContext]]
+         like: `on_partition_initialize(partition_context)`.
+        :paramtype on_partition_initialize: Callable[[~azure.eventhub.aio.PartitionContext]]
         :keyword on_partition_close: The callback function that will be called after a consumer for a certain
-         partition is closed. The callback takes two parameters: `partition_context` which contains partition
-         information and `reason` for the close. The callback should be defined like so:
+         partition is closed. It would be also called when error is raised during receiving after retry attempts
+         exhaust. The callback takes two parameters: `partition_context` which contains partition
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: "attempts are exhausted"

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_client_async.py [hunk 14]
```diff
@@ -64,8 +64,14 @@ class EventHubConsumerClient(ClientBaseAsync):
     :keyword float auth_timeout: The time in seconds to wait for a token to be authorized by the service.
      The default value is 60 seconds. If set to 0, no timeout will be enforced from the client.
     :keyword str user_agent: The user agent that should be appended to the built-in user agent string.
-    :keyword int retry_total: The total number of attempts to redo a failed operation when an error occurs. Default
-     value is 3.
+    :keyword int retry_total: The total number of attempts to redo a failed operation when an error occurs.
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_client_async.py]: Same comments as on the sync fle

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py (hunk 8 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: In current implementation, if the user doesn't provide a checkpoint_store, we'd internally use a InMemoryCheckpointStore which is implicit to user.

So previously when user call update_checkpoint without giving checkpoint_store, we would log an warning.
Now the update_checkpoint can work without a user provided checkpoint_store.

Maybe something like:

When the checkpoint_store is not provided, the SDK will use a InMemoryCheckpointStore internally which can do in-memory checkpoint .

Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: In current implementation, if the user doesn't provide a checkpoint_store, we'd internally use a InMemoryCheckpointStore which is implicit to user.

So previously when user call update_checkpoint without giving checkpoint_store, we would log an warning.
Now the update_checkpoint can work without a user provided checkpoint_store.

Maybe something like this?
```
When the checkpoint_store is not provided, the SDK will use a InMemoryCheckpointStore internally which can do in-memory checkpoint .
```

Or do you have any better ideas..

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py (hunk 7 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: thanks, updated

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py (hunk 9 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: thanks, updated

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py (hunk 10 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: thanks, updated

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py (hunk 11 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: thanks, updated

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py (hunk 12 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: thanks, updated

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py (hunk 13 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: thanks, updated

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_client_async.py (hunk 14 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_client_async.py]: thanks, updated

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py (hunk 8 above)
Comment (annatisch) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py]: Ah - in that case I would go with something like:
```
If a checkpoint store is not provided, the checkpoint will be maintained internally in memory.
```
